packages/sanity/src/desk/panes/document/documentPanel/header/DocumentPanelHeader.tsx

```diff
@@ -98,14 +101,9 @@ export const DocumentPanelHeader = memo(
               </>
             )}
 
-            {validation.length > 0 && (
-              <ValidationMenu
-                boundaryElement={rootElement}
-                isOpen={isValidationOpen}
-                key="validation-menu"
-                setOpen={setValidationOpen}
-              />
-            )}
+            {actionItems.map((item, itemIndex) => (
```

Comment:
Possibly an edge case: but I configured 100 inspectors, and the UI can't handle that many inspectors. I wonder if we could solve it by rendering the inspector buttons in a menu when they exceed a certain number (not the validation inspector). Something similar to the image below (not sure which icon might be appropriate, I used the `RobotIcon` as a placeholder icon).

<img width="785" alt="Screenshot 2023-05-26 at 17 08 28" src="https://github.com/sanity-io/sanity/assets/15094168/fc2767bc-38a0-453e-9497-f4aaf90d6985">

Reply:
I realized that this can be handled by setting `showAsAction` to `false`!